### DISCUSSION TITLE
Telegram Integration

### DISCUSSION THREAD

**Discussion by gnulab-id** [DigitalRuby/IPBan#182] [Ideas]
Hi,

Telegram nowadays is pretty popular with notifications by using their bot service. So, would it be possible to integrate IPBan with Telegram?

Say, for every login failure and the associated IP got blacklisted, such action will send a notification to Telegram?

#1 2022-03-25 00:06:00.7012|WARN|DigitalRuby.IPBanCore.Logger|Banning ip address: 188.68.33.172, user name: ADMINISTRATOR, config black listed: True, count: 3, extra info: , duration: 1.00:00:00
#2 2022-03-25 01:56:21.3234|WARN|DigitalRuby.IPBanCore.Logger|Banning ip address: 87.127.198.86, user name: ADMINISTRATOR, config black listed: True, count: 3, extra info: , duration: 1.00:00:00

#1 , #2 would be appended for the user to know whether it is high traffic for the day or a slow walk in the park day.

My situation is very simple. I have a VPS installed with Windows 10 and has RDP activated. I login to this RDP occasionally, once every few days. It is quite a chore just login from desktop to see the number of failed login attempts for the day. Plus, I have all my mikrotiks send failed login attempts to Telegram already.

Sending notification via email? Yeah, can do that, but somehow via Telegram seems to be cooler.

Thanks
Henry

**Comment by jjxtra** [comment]
Running external process on ban is simple. So you could run your script
that posts to telegram. Let me know if the processtorunonban field in the
config file makes sense.

On Thu, Mar 24, 2022 at 6:58 PM gnulab-id ***@***.***> wrote:

> Hi,
>
> Telegram nowadays is pretty popular with notifications by using their bot
> service. So, would it be possible to integrate IPBan with Telegram?
>
> Say, for every login failure and the associated IP got blacklisted, such
> action will send a notification to Telegram?
>
> #1 <https://github.com/DigitalRuby/IPBan/issues/1> 2022-03-25
> 00:06:00.7012|WARN|DigitalRuby.IPBanCore.Logger|Banning ip address:
> 188.68.33.172, user name: ADMINISTRATOR, config black listed: True, count:
> 3, extra info: , duration: 1.00:00:00
> #2 <https://github.com/DigitalRuby/IPBan/issues/2> 2022-03-25
> 01:56:21.3234|WARN|DigitalRuby.IPBanCore.Logger|Banning ip address:
> 87.127.198.86, user name: ADMINISTRATOR, config black listed: True, count:
> 3, extra info: , duration: 1.00:00:00
>
> #1 <https://github.com/DigitalRuby/IPBan/issues/1> , #2
> <https://github.com/DigitalRuby/IPBan/issues/2> would be appended for the
> user to know whether it is high traffic for the day or a slow walk in the
> park day.
>
> My situation is very simple. I have a VPS installed with Windows 10 and
> has RDP activated. I login to this RDP occasionally, once every few days.
> It is quite a chore just login from desktop to see the number of failed
> login attempts for the day. Plus, I have all my mikrotiks send failed login
> attempts to Telegram already.
>
> Sending notification via email? Yeah, can do that, but somehow via
> Telegram seems to be cooler.
>
> Thanks
> Henry
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/DigitalRuby/IPBan/discussions/182>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJWQGAC23L4J5TBIPGUNV3VBUFUNANCNFSM5RSXMHAQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
-- Jeff
